mkdocs.yml

```diff
@@ -1,21 +1,51 @@
 docs_dir: docs/book
 site_dir: docs/html
+extra:
+    project: Mezzio
+    current_version: v2
+    versions:
+      - v2
+      - v1
 nav:
     - Home: index.md
-    - Introduction: intro.md
-    - Reference:
-      - "HAL": hal.md
-      - "Links and Resources": links-and-resources.md
-      - "Generating Resources": resource-generator.md
-      - "Custom Resources": cookbook/generating-custom-resources.md
-      - "Representations": representations.md
-      - "Generating HAL for Doctrine Entities": doctrine.md
-      - "Factories": factories.md
-    - Cookbook:
-        - "Generating Custom Links In Middleware and Request Handlers": cookbook/generating-custom-links-in-middleware.md
-        - "Using the ResourceGenerator in path-segregated middleware": cookbook/path-segregated-uri-generation.md
+    - v2:
+      - Introduction: v2/index.md
+      - Reference:
+        - "HAL": v2/hal.md
+        - "Links and Resources": v2/links-and-resources.md
+        - "Generating Resources": v2/resource-generator.md
+        - "Custom Resources": v2/cookbook/generating-custom-resources.md
+        - "Representations": v2/representations.md
+        - "Generating HAL for Doctrine Entities": v2/doctrine.md
+        - "Factories": v2/factories.md
+        - "Migration from V1": v2/migration.md
```

Comment:
```suggestion
    - Migration:
        - "Migration from V1": v2/migration.md
```

Reply:
Why as a separate section? We generally will have one migration guide per version (v3 will have a "migration from v2", but will not have a "migration from v1" document). I'm not sure it justifies having a section separate from the reference.

Reply:
Main reason: consistency. We can have migration guides for minor versions in the future – in this component or others.